### PR TITLE
deps(ember): upgrade app to ember@4.12 + export of ember-modifier as direct dependency

### PR DESCRIPTION
**Pull request by jayjayjpg**
Part of https://github.com/meroxa/platform-ui-v1/issues/881
Part of https://github.com/meroxa/product/issues/749
Closes https://github.com/ConduitIO/mx-ui-components/issues/51

With this change, we're

- upgrading mx-ui-components to the latest version `ember@4.12`
- exporting `ember-modifier` as a direct dependency as required for the upgrade of our host application

Once this is merged and released, the upgrade work in https://github.com/meroxa/platform-ui-v1/issues/881 will be unblocked.

## Demo

Chromatic Build: https://www.chromatic.com/build?appId=6109b9596e1c1300390c39c5&number=241

### CODE REVIEW

**Review by janelletavares — APPROVED** [review]
looks reasonable to me

File: package.json [hunk 1]
```diff
@@ -35,22 +37,26 @@
     "build-storybook": "yarn run build:prod --"
   },
   "dependencies": {
-    "ember-auto-import": "^2.5.0",
+    "ember-auto-import": "^2.6.3",
     "ember-cli-babel": "^7.26.11",
-    "ember-cli-htmlbars": "^6.1.1",
+    "ember-cli-htmlbars": "^6.2.0",
     "ember-keyboard": "8.2.0",
+    "ember-modifier": "^4.1.0",
```
Comment (jmar910) [inline package.json:44]: this is nice

File: .prettierrc.js [hunk 2]
```diff
@@ -3,6 +3,12 @@
 module.exports = {
   singleQuote: true,
   overrides: [
+    {
+      files: '*.{js,ts}',
+      options: {
+        singleQuote: true,
+      },
+    },
```
Comment (raulb) [inline .prettierrc.js:11]: Nice addition.

**Review by raulb — APPROVED** [review]
Love these changes.

**Pull request merged**